client-side/src/ProxyKey.tsx

```diff
@@ -8,38 +8,61 @@ interface Props {
     keyEntity: Key
     useAltStyle: boolean
     onKeyRevoked(remainingKeys: Key[]): void
+    onKeyReinstated(remainingKeys: Key[]): void
 }
 
-const ProxyKey: FC<Props> = ({ keyEntity: key, useAltStyle, onKeyRevoked }) => {
+const ProxyKey: FC<Props> = ({ keyEntity: key, useAltStyle, onKeyRevoked, onKeyReinstated }) => {
     const app = useContext(AppContext);
     const conversion = useConversion();
 
     const [revoking, setRevoking] = useState<boolean>(false);
+    const [reinstating, setReinstating] = useState<boolean>(false);
 
     const revoke = async (event: React.MouseEvent) => {
         event.preventDefault();
 
         setRevoking(true);
-        const response = await fetch(`${app.serverBaseUrl}/api/keys?key=${key.key}`, {
-            method: 'delete',
+        const response = await fetch(`${app.serverBaseUrl}/api/keys?key=${key.key}&action=revoke`, {
+            method: 'update',
```

Comment:
This should be an http verb, so `patch` or `put` would work. The most RESTful way of revoking/reinstating would be to pass the entire key model with the `revoked` property set to either a Date or null.